homework/index.js

```diff
@@ -30,15 +32,138 @@
+    });
+    const repoName = document.querySelector('.repository-data');
+    createAndAppend('a', repoName, {
+      href: hyfRepo.html_url,
+      target: '_blank',
+      text: hyfRepo.name,
+      class: 'repo-name',
+    });
+    if (hyfRepo.description === null) {
+      document.querySelector('.description-row').setAttribute('class', 'hide-description-row');
+    } else {
+      document.querySelector('.description-data').innerText = hyfRepo.description;
+    }
+    document.querySelector('.forks-data').innerText = hyfRepo.forks;
+    document.querySelector('.updated-data').innerText = new Date(
+      hyfRepo.updated_at,
+    ).toLocaleString();
+  }
+
+  // Creating the contributions part
+  async function createContributors(rightDiv, url) {
+    createAndAppend('h3', rightDiv, {
+      text: 'Contributions: ',
+      class: 'contributor-header',
+    });
+    try {
+      const contributors = await fetchJSON(url);
+      if (!(contributors && contributors.length)) {
+        createAndAppend('div', rightDiv, {
+          text: 'No contributions found.',
+          class: 'alert-error',
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const ul = createAndAppend('ul', rightDiv, {
+          class: 'contributor-list',
+        });
+        contributors.forEach(contributor => {
+          const contributorItem = createAndAppend('li', ul, {
+            class: 'contributor-item',
+          });
+          const contributorLink = createAndAppend('a', contributorItem, {
+            href: contributor.html_url,
+            target: '_blank',
+          });
+          const contributorDiv = createAndAppend('div', contributorLink, {
+            class: 'contributor',
+          });
+          createAndAppend('img', contributorDiv, {
+            src: contributor.avatar_url,
+            class: 'contributor-avatar',
+          });
+          const contributorData = createAndAppend('div', contributorDiv, {
+            class: 'contributor-data',
+          });
+          createAndAppend('div', contributorData, {
+            text: contributor.login,
+            class: 'contributor-login',
+          });
+          createAndAppend('div', contributorData, {
+            text: contributor.contributions,
+            class: 'contributor-badge',
+          });
+        });
       }
+    } catch (err) {
+      createAndAppend('div', rightDiv, {
+        text: err.message,
+        class: 'alert-error',
+      });
+    }
+  }
+
+  async function main(url) {
+    const root = document.getElementById('root');
+
+    // Creating the header and the select menu
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('h2', header, {
+      text: 'HYF Repositories ',
+      class: 'nav-title',
     });
+    const selectMenu = createAndAppend('select', header, {
+      id: 'repo-selector',
+    });
+    try {
```

Comment:
Although this logic works just fine, I would suggest that you split your `main` function a bit further as well. Currently, it is doing multiple things like creating the layout for header, fetching repository details and creating the dropdown. I would suggest that you can also move fetching the repository details in a separate function outside and call that from `main`.